src/main/java/seedu/address/logic/commands/DeleteContactEventCommand.java

```diff
@@ -0,0 +1,123 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.model.Model.PREDICATE_SHOW_ALL_PERSONS;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Set;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.calendar.Calendar;
+import seedu.address.model.event.Event;
+import seedu.address.model.event.exceptions.EventNotFoundException;
+import seedu.address.model.person.Address;
+import seedu.address.model.person.Email;
+import seedu.address.model.person.Name;
+import seedu.address.model.person.Person;
+import seedu.address.model.person.Phone;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Delete an event from the calendar of an existing person in the address book.
+ */
+public class DeleteContactEventCommand extends Command {
+
+    public static final String COMMAND_WORD = "deleteContactEvent";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Deletes an event from the calendar of the person "
+            + "identified by the index number used in the displayed person list.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + "ANY TIME WITHIN EVENT DURATION \n"
+            + "Example: " + COMMAND_WORD + " "
+            + "2024-01-01 12:00 ";
+
+    public static final String MESSAGE_DELETE_EVENT_FROM_PERSON_SUCCESS = "Event deleted from %s: %2$s";
+    public static final String MESSAGE_NO_EVENT = "There is no valid existing event at this timing.";
+
+    private final Index index;
+    private final LocalDateTime eventTime;
+
+    /**
+     * @param index of the person in the filtered person list to delete an event from.
+     * @param eventTime time of the event to delete.
+     */
+    public DeleteContactEventCommand(Index index, LocalDateTime eventTime) {
+        requireNonNull(index);
+        requireNonNull(eventTime);
+
+        this.index = index;
+        this.eventTime = eventTime;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Person> lastShownList = model.getFilteredPersonList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
```

Comment:
Consider using a try block that catches IndexOutOfBoundsException instead

Reply:
Good idea here too, resolved as well.